presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/json/format/CustomDateTimeFormatFunctions.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.kafka.encoder.json.format;
+
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.joda.time.format.DateTimeFormat;
+import org.joda.time.format.DateTimeFormatter;
+
+import java.util.Locale;
+
+import static io.prestosql.spi.type.DateTimeEncoding.unpackMillisUtc;
+import static io.prestosql.spi.type.DateTimeEncoding.unpackZoneKey;
+import static java.lang.String.format;
+
+public class CustomDateTimeFormatFunctions
+{
+    private CustomDateTimeFormatFunctions() {}
+
+    public static JsonDateTimeFormatFunction dateFormatFunction(String formatHint)
+    {
+        try {
+            DateTimeFormatter formatter = DateTimeFormat.forPattern(formatHint).withLocale(Locale.ENGLISH);
+            return (millis -> (new DateTime(millis, DateTimeZone.UTC)).toString(formatter));
```

Comment:
The parentheses around the return value are not necessary (and I personally feel they reduce readability).  I also like more passing the temporal value to the formatter:
`return millis -> formatter.print(new DateTime(millis, UTC));`

A possible optimization could be pre-instantiating formatters for all patterns when creating the encoder and reusing them (the Joda ones are thread safe), as opposed to doing it for every row.

Reply:
Shouldn't these functions reuse the same `DateTimeFormatter's`?

Reply:
You're right, you're actually doing what I was suggesting.  I misread the code.